File: clients/haskell/src/RAIC/Model/UnitAction.hs
module RAIC.Model.UnitAction where

import           GHC.Generics          (Generic)
import           RAIC.Model.Vec2Double (Vec2Double)
import           RAIC.Utils.Trans      (Trans)

data UnitAction = UnitAction {
  velocity    :: Double,
  jump        :: Bool,
  jump_down   :: Bool,
  aim         :: Vec2Double,
  shoot       :: Bool,
  reload      :: Bool,
  swap_weapon :: Bool,
  plant_mine  :: Bool
} deriving (Generic, Show)

instance Trans UnitAction

File: clients/rust/model/src/unit_action.rs
use crate::*;
#[derive(Clone, Debug, trans::Trans)]
pub struct UnitAction {
    pub velocity: f64,
    pub jump: bool,
    pub jump_down: bool,
    pub aim: Vec2F64,
    pub shoot: bool,
    pub reload: bool,
    pub swap_weapon: bool,
    pub plant_mine: bool,
}
